pyproject.toml

```diff
@@ -68,12 +67,12 @@ grpc = ["grpcio"]
 build-develop = "python scripts/setup_bridge.py develop"
 build-develop-with-release = { cmd = "python scripts/setup_bridge.py develop", env = { TEMPORAL_BUILD_RELEASE = "1" }}
 fix-wheel = "python scripts/fix_wheel.py"
-format = [{cmd = "black ."}, {cmd = "isort ."}]
+format = [{cmd = "ruff check --target-version py38 --select I --fix"}, {cmd = "ruff format --target-version py38"}, ]
 gen-docs = "python scripts/gen_docs.py"
 gen-protos = "python scripts/gen_protos.py"
 lint = [
-  {cmd = "black --check ."},
-  {cmd = "isort --check-only ."},
+  {cmd = "ruff check --target-version py38 --select I"},
+  {cmd = "ruff format --check"},
```

Comment:
In theory `target-version` should be able to go in a `[tool.ruff]` section. However when I did that, it did not appear to pass the option.

In contrast, here's what it prints out with the option passed explicitly (current PR state):

```
(temporalio-py3.8) sdk-python(ruff) poe format
Poe => ruff check --target-version py38 --select I --fix
All checks passed!
Poe => ruff format --target-version py38
267 files left unchanged
```